### ISSUE TITLE
Concatenated tuples raise tuple size mismatch error

### ISSUE THREAD

**Issue by Trezorro** [microsoft/pyright#2075]
**Describe the bug**
When passing a concatenated tuple to a function with a specific tuple length signature, the following error is raised (via Pylance in my case):
```
Argument of type "Tuple[int, ...]" cannot be assigned to parameter "four" of type "Tuple[int, int, int, int]" in function "recieve_four"
  Tuple size mismatch; expected 4 but received indeterminate number
```

**To Reproduce**
A minimal code snippet to reproduce. In Vscode, with Pylance and `"python.analysis.typeCheckingMode": "basic",` 
```python
from typing import Tuple

def recieve_four(four: Tuple[int,int,int,int]):
    print(len(four)==4)

recieve_four((1,2)+(3,4))
```
**Expected behavior**
Either the size of the concatenated should be determined, or `Tuple[int, ...]` should satisfy the `Tuple[int, int, int, int]` signature.

**Additional context**
Such notation with concatenating tuples is used in the context of coordinates, such as in the PIL documentation https://pillow.readthedocs.io/en/stable/reference/ImageDraw.html#example-draw-a-gray-cross-over-an-image

**Comment by erictraut** [comment]
The addition operator for tuples is defined in `builtins.pyi` as the following:
```python
    def __add__(self, x: Tuple[_T, ...]) -> Tuple[Union[_T_co, _T], ...]: ...
```

Based on this type information, pyright is doing the right thing here.

To address this, the type checker would need to have specialized (hard-coded) knowledge of how the `tuple` class implements the `__add__` method. We're generally reluctant to add such special-case knowledge, even for built-in types, because this approach doesn't scale.

On the other hand, mypy (another Python type checker) appears to have implemented this special case.

I'll need to think more about this feature request — and investigate whether it can be implemented in a way that doesn't potentially slow down common code paths in the analyzer.

**Comment by erictraut** [comment]
Since mypy handles this special case, I've decided to support it also. This will be in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.156, which I just published. It will also be included in the next release of pylance.
